example/client.rb

```diff
@@ -68,7 +68,9 @@
 
 stream.on(:close) do
   log.info 'stream closed'
-  sock.close
+
+  # keep connection open to handle push messages
+  # sock.close
```

Comment:
:close callback is called when the other end closes the stream.. At this point, we can't push against this stream anymore. Keep the sock.close as is :)